server/controllers/profile.js

```diff
@@ -38,3 +40,39 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
     },
   });
 });
+
+// @route POST /profile/upload-image
+// @desc upload user's image
+// @access Private
+exports.uploadProfileImage = asyncHandler(async (req, res, next) => {
+  const id = req.user.id;
+
+  singleUpload(req, res, function (err) {
+    if (err) {
+      return res.json({
+        success: false,
+        errors: {
+          title: "Image Upload Error",
+          detail: err.message,
+          error: err,
+        },
+      });
+    }
+
+    const update = { photo: req.file.location };
+
+    try {
```

Comment:
You don't need to use try/catch as the asyncHandler handles promise rejections and passes it to the error middleware :)